modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/persistence/RocksDbKeyValueStorage.java

```diff
@@ -1281,8 +1273,6 @@ private Set<UpdateOnlyRevisionEvent> collectUpdateRevisionEventsFromStorage(long
 
     @Override
     public HybridTimestamp timestampByRevision(long revision) {
-        rwLock.readLock().lock();
-
         try {
             assertRequestedRevisionLessThanOrEqualToCurrent(revision, rev);
```

Comment:
We are accessing a non-volatile field here, are you sure this is safe?

Reply:
No, made it volatile